### PR TITLE
Makes organ harvester drop circuitboard on deconstruct.

### LINKED ISSUE #75160: Organ harvester doesn't drop a circuitboard when deconstructed
roundid 205034

build a organ harvester
deconstruct it
there is no circuitboard

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Closes https://github.com/tgstation/tgstation/issues/75160
Calls parent on the /exited proc since it was being overridden when it needs parent.
## Why It's Good For The Game
Bugfix
## Changelog
:cl:
fix: Organ harvester board drops on deconstruct.
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
I'm confused. Since you are applying this fix to `_machinery.dm` file shouldn't that mean all machines don't drop circuit boards when deconstructed? why is this problem unique only to organ harvester? Something doesn't add up

**Comment by SyncIt21** [comment]
Ohhh i get it now, It's a big indirection, in `deconstruct()` It start's when we move the circuit board out of the machine here
https://github.com/tgstation/tgstation/blob/046a5cbe01e5c8a06337b7fbdc376e4765e94b18/code/game/machinery/_machinery.dm#L817

Since the circuit board is part of the machine's contents, when it gets moved out it triggers the machines `Exited()` proc. This nulls the circuit board and removes it from the machines component parts
https://github.com/tgstation/tgstation/blob/046a5cbe01e5c8a06337b7fbdc376e4765e94b18/code/game/machinery/_machinery.dm#L1160-L1162

So that's why all machines properly deconstruct except for the organ harvester because the organ harvester overrides the `Exited()` but it doesn't call it's super proc
https://github.com/tgstation/tgstation/blob/046a5cbe01e5c8a06337b7fbdc376e4765e94b18/code/game/machinery/harvester.dm#L192-L195
It's not calling `return ..()`. So this is the actual fix, make sure to add that along with your changes as well because we don't want to qdel an null object.

I don't like the way this was set up with all this indirection when it should have been a simple fix, maybe we should just not null the circuit when it's moved out of the machine because this changes the length of this for loop.

https://github.com/tgstation/tgstation/blob/046a5cbe01e5c8a06337b7fbdc376e4765e94b18/code/game/machinery/_machinery.dm#L811

mid iteration and I already had this problem with the lighting subsystem PR. @LemonInTheDark maybe you have experience with this?

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -198,14 +198,16 @@
...
-	if (!isnull(component_parts))
+	if(!isnull(component_parts))
 		// Don't delete the stock part singletons
 		for (var/atom/atom_part in component_parts)
 			qdel(atom_part)
 		component_parts.Cut()
 		component_parts = null
 
-	QDEL_NULL(circuit)
+	if(!isnull(circuit))
```
Comment (Mothblocks) [inline code/game/machinery/_machinery.dm:208]: you do not need this check, QDEL_NULL(null) is not only benign but something lemon wants to make faster in the future

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Because `_machinery.dm` now has no meaningful changes, I'd like for you to not touch it in this PR

**Comment by Xander3359** [comment]
Did I fuck up checks? It looks like messages or something is breaking but I haven't touched that at all.

**Comment by Mothblocks** [comment]
Merge skew, not your fault

**Pull request merged**